src/Workspaces/CSharp/Portable/Formatting/Rules/NewLineUserSettingFormattingRule.cs

```diff
@@ -370,6 +379,19 @@ public override AdjustNewLinesOperation GetAdjustNewLinesOperation(SyntaxToken p
                 }
             }
 
+            // * { - in the switch expression context
+            if (currentToken.Kind() == SyntaxKind.OpenBraceToken && currentToken.Parent != null && currentToken.Parent.IsKind(SyntaxKind.SwitchExpression))
+            {
+                if (optionSet.GetOption(CSharpFormattingOptions.NewLinesForBracesInControlBlocks))
+                {
+                    return CreateAdjustNewLinesOperation(1, AdjustNewLinesOption.PreserveLines);
```

Comment:
❔ Why is this using `PreserveLines` and not `ForceLinesIfOnSingleLine`. I see both approaches in the surrounding code but the difference is not clear to me.

Reply:
I actually don't know what the difference is. I looked at how this works for the control blocks (`unsafe`, `if`, etc.), and took that approach.

I'll see if I can look into this, and explain if, and what the difference is.

Reply:
Ok, just looked into to this, but cannot really make any sense of how this _exactly_ works. I've been playing around a bit with this option, and a couple scenario's that might trigger something here. For example, when I change the option to `ForceLinesIfOnSingleLine` on the given scenario then nothing changes, which is also what you'd expect (it will still apply a single line, since it's `...OnSingleLine`).

However, when you change the scenario to something like:

```csharp
var msg = 1 switch


{
    _ => null
};
```

Then I'd expect `PreserveLines` to maintain the lines between the `switch` keyword, and the opening brace, which is _what happens`. However, even with `ForceLinesIfOnSingleLine` I see identical behavior :( So I cannot really make any sense of how this is consumed exactly.

I might dig into this a bit further, but don't hold your breath for it ;)

Reply:
Is there a difference in the two for this?

```csharp
var msg = 1 switch { _ => null };
```

Thanks for looking into it the amount you did 😄

Reply:
No, there isn't. I just checked it now, but with all 4 branches (default vs. non-default, and `PreserveLines` vs `ForceLinesIfOnSingleLine`). It puzzles me how this supposes to work...